### PR TITLE
Add a workaround for hanging proxy in nodeProxyRequest

### PR DESCRIPTION
**Pull request by gmarek**
cc @lavalamp @yujuhong

### CODE REVIEW

File: test/e2e/kubelet_stats.go [hunk 1]
```diff
@@ -339,28 +344,38 @@ type usageDataPerContainer struct {
 
 // Performs a get on a node proxy endpoint given the nodename and rest client.
 func nodeProxyRequest(c *client.Client, node, endpoint string) (restclient.Result, error) {
+	// proxy tends to hang in some cases when Node is not ready. Add an artificial timeout for this call. This will leak a goroutine if proxy hangs.
```
Comment (wojtek-t) [inline test/e2e/kubelet_stats.go]: Can you please link the issue that you created for it?

**Comment by wojtek-t** [comment]
One minor nit

File: test/e2e/kubelet_stats.go (hunk 1 above)
Comment (gmarek) [inline test/e2e/kubelet_stats.go]: Done.

**Comment by wojtek-t** [comment]
LGTM

**Comment by lavalamp** [comment]
Did we make an issue for fixing the proxy timeout? I can't find it. We should note this there so we can revert it when fixed.

**Comment by wojtek-t** [comment]
@lavalamp - the issue is: https://github.com/kubernetes/kubernetes/issues/22165

**Pull request merged**